frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -308,27 +308,30 @@ end
 
 function ReaderRolling:onCheckDomStyleCoherence()
     if self.ui.document and self.ui.document:isBuiltDomStale() then
-        local has_bookmarks_warn_txt = ""
         -- When using an older DOM version, bookmarks may break
-        if self.using_non_normalized_xpointers and self.ui.bookmark:hasBookmarks() then
-            has_bookmarks_warn_txt = _("\nNote that this change in styles may render your bookmarks or highlights no more valid.\nIf some of them do not show anymore, you can just revert the change you just made to have them shown again.\n\n")
-        end
-        UIManager:show(ConfirmBox:new{
-            text = T(_("Styles have changed in such a way that fully reloading the document may be needed for a correct rendering.\n%1Do you want to reload the document?"), has_bookmarks_warn_txt),
-            ok_callback = function()
-                -- Allow for ConfirmBox to be closed before showing
-                -- "Opening file" InfoMessage
-                UIManager:scheduleIn(0.5, function()
-                    -- And check we haven't quit reader in these 0.5s
-                    if self.ui.document then
-                        self.ui:reloadDocument()
-                    end
-                end)
-            end,
-        })
+        local bookmarks_warn_txt = self.using_non_normalized_xpointers and self.ui.annotation:hasAnnotations()
+            and _("\nNote that this change in styles may render your bookmarks or highlights no more valid.\nIf some of them do not show anymore, you can just revert the change you just made to have them shown again.\n\n")
+        self:showReloadConfirmBox(bookmarks_warn_txt)
     end
 end
 
+function ReaderRolling:showReloadConfirmBox(warn_txt)
```

Comment:
May be `showSuggestReloadConfirmBox()` ? as it has some hardcoded message in that sense, it's not a generic reload document thingie.